lighthouse-core/lib/network-request.js

```diff
@@ -246,6 +247,18 @@ module.exports = class NetworkRequest {
 
     this.responseReceivedTime = timestamp;
 
+    this.responseHeadersText = response.headersText || '';
+    this.responseHeaders = NetworkRequest._headersDictToHeadersArray(response.headers);
+
+    // The total length of the encoded data is spread out among multiple events. The sum of the
+    // values in onResponseReceived and all the onDataReceived events will equal the value
+    // seen on the onLoadingFinished event. As we process onResonseReceived and onDataReceived
```

Comment:
> The sum of the
>    // values in onResponseReceived and all the onDataReceived events will equal the value
>    // seen on the onLoadingFinished event. 

I don't know exactly what leads to a situation when the loadingFinished total is different than the sum of the parts, but apparently it does happen.

I just tried against 6 saved devtoolslogs i have. (Disclaimer: the artifacts were collected months ago definitely without networkservice, though I wouldn't expect it to matter)... There's, 320 network requests in total and _nearly all_ of them have a sum that matches (or the sum was 0).. But I found 3 requests where this wasn't the case.

| encodedSum | encodedTotalInLoadingFinished | url |
|-|-|-|
| 108 | 12640 |  http://localhost:10200/dobetterweb/dbw_tester.html |
| 139 | 144 |  http://localhost:10200/dobetterweb/empty_module.js?delay=500 |
| 650 | 714 |  https://s.amazon-adsystem.com/x/da2e6c890e6e3636 |

Again.. don't know what leads to this situation, but it happens. I'm happy with admitting that in the comment or we can ask caseq/network people about the circumstances..